launchpad-common/src/winner_selection.rs

```diff
@@ -193,4 +195,12 @@ pub trait WinnerSelectionModule:
 
         ticket_ids
     }
+
+    fn check_caller_owner_or_user(&self) {
+        if self.blockchain().get_owner_address() == self.blockchain().get_caller() {
+            return;
+        }
+
+        self.blockchain().check_caller_is_user_account();
+    }
```

Comment:
This doesn't respect the doc, i.e. `or an account from another shard`, you just check the caller is a user account.

Reply:
Indeed, this was not what was requested, but it is a more simpler and clear/cleaner solution. which covers the initial request. No need for external SCs to call this function.